### DISCUSSION TITLE
Can I use `wrangler dev` with "offline" local KV namespaces?

### DISCUSSION THREAD

**Discussion by oxcafedead** [cloudflare/workers-sdk#3553] [Q&A] [answered]
The idea is that I don't want to create some KV namespaces in Cloudflare remote servers. I want to work in a local sandbox.

When I do `wrangler dev` locally (latest 3x wrangler), I get
```
X [ERROR] In development, you should use a separate kv namespace than the one you'd use in production. Please create a new kv namespace with "wrangler kv:namespace create <name> --preview" and add its id as preview_id to the kv_namespace "xxx" in your wrangler.toml
```
If I try to do `wrangler kv:namespace create xxx --preview`, it tries to authenticate me in Cloudflare. If I do it, it creates *a real* "preview" KV. I don't want this to happen, only local small in-memory KV. 
Is there any way to start local KV without publishing it?

The end goal: to create an isolated Docker container with my worker which can be used as a standalone dependency.

**Comment by oxcafedead** [comment]
This may probably be related to https://github.com/cloudflare/workers-sdk/issues/2462 but I am not sure

**Comment by penalosa** [comment, marked as answer]
In local mode (`wrangler dev` in v3, `wrangler dev --local` in v2) no remote resources are accessed. You can copy the ID used in your namespace ID to the preview ID field in wrangler.toml, which should work locally.

**Reply by oxcafedead** [reply]
Indeed, this was my silly error (the placeholders were not filled in my `wrangler.toml` template). Thanks.
